app/src/androidTest/java/org/mozilla/fenix/ui/NoNetworkAccessStartupTests.kt

```diff
@@ -0,0 +1,109 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ui
+
+import androidx.core.net.toUri
+import androidx.test.espresso.IdlingRegistry
+import org.junit.After
+import org.junit.Test
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.HomeActivityTestRule
+import org.mozilla.fenix.helpers.TestHelper.packageName
+import org.mozilla.fenix.helpers.TestHelper.setNetworkConnection
+import org.mozilla.fenix.helpers.TestHelper.verifyUrl
+import org.mozilla.fenix.helpers.idlingresource.NetworkConnectedIdlingResource
+import org.mozilla.fenix.helpers.idlingresource.NetworkDisconnectedIdlingResource
+import org.mozilla.fenix.ui.robots.browserScreen
+import org.mozilla.fenix.ui.robots.homeScreen
+import org.mozilla.fenix.ui.robots.navigationToolbar
+
+/**
+ * Tests to verify some main UI flows with Network connection off
+ *
+ */
+
+class NoNetworkAccessStartupTests {
+    private val activityTestRule = HomeActivityTestRule()
+    private val networkDisconnectedIdlingResource = NetworkDisconnectedIdlingResource()
+
+    @After
+    fun tearDown() {
+        setNetworkConnection(true)
+        IdlingRegistry.getInstance().unregister(NetworkConnectedIdlingResource())
```

Comment:
The docs don't make it clear if this will actually unregister the resource: it's not clear if we can use a separate instance when registering or unregistering.

Reply:
done